zirgen/dsl/driver.cpp

```diff
@@ -571,7 +572,6 @@ int main(int argc, char* argv[]) {
   }
 
   pm.addPass(mlir::createCanonicalizerPass());
-  pm.addPass(mlir::createCSEPass());
```

Comment:
We might still want to run CSE at this point — since everything gets inlined and unrolled in the CheckFunc, I expect there should be substantial benefit to running it here. My intuition says that early CSE shrinks the code before "the big blowup" of generating the validity stuff, so the code here should be smaller than before. Are the compile times still expensive with both of them, or does the early CSE make the second one cheaper?

Reply:
Compile times are still expensive with both passes; nothing, so far as I can tell, makes the second pass cheaper.